M/mlpack/build_tarballs.jl

```diff
@@ -0,0 +1,175 @@
+    # We need to rename some functions for compatibility with Julia's OpenBLAS
+    SYMB_DEFS=()
+    for sym in cgbcon cgbsv cgbsvx cgbtrf cgbtrs cgecon cgees cgeev cgeevx cgehrd cgels cgelsd cgemm cgemv cgeqrf cgesdd cgesv cgesvd cgesvx cgetrf cgetri cgetrs cgges cggev cgtsv cgtsvx cheev cheevd cherk clangb clange clanhe clansy cpbtrf cpocon cposv cposvx cpotrf cpotri cpotrs ctrcon ctrsyl ctrtri ctrtrs cungqr dasum ddot dgbcon dgbsv dgbsvx dgbtrf dgbtrs dgecon dgees dgeev dgeevx dgehrd dgels dgelsd dgemm dgemv dgeqrf dgesdd dgesv dgesvd dgesvx dgetrf dgetri dgetrs dgges dggev dgtsv dgtsvx dlahqr dlangb dlange dlansy dlarnv dnrm2 dorgqr dpbtrf dpocon dposv dposvx dpotrf dpotri dpotrs dstedc dsyev dsyevd dsyrk dtrcon dtrevc dtrsyl dtrtri dtrtrs ilaenv sasum sdot sgbcon sgbsv sgbsvx sgbtrf sgbtrs sgecon sgees sgeev sgeevx sgehrd sgels sgelsd sgemm sgemv sgeqrf sgesdd sgesv sgesvd sgesvx sgetrf sgetri sgetrs sgges sggev sgtsv sgtsvx slahqr slangb slange slansy slarnv snrm2 sorgqr spbtrf spocon sposv sposvx spotrf spotri spotrs sstedc ssyev ssyevd ssyrk strcon strevc strsyl strtri strtrs zgbcon zgbsv zgbsvx zgbtrf zgbtrs zgecon zgees zgeev zgeevx zgehrd zgels zgelsd zgemm zgemv zgeqrf zgesdd zgesv zgesvd zgesvx zgetrf zgetri zgetrs zgges zggev zgtsv zgtsvx zheev zheevd zherk zlangb zlange zlanhe zlansy zpbtrf zpocon zposv zposvx zpotrf zpotri zpotrs ztrcon ztrsyl ztrtri ztrtrs zungqr; do
+        SYMB_DEFS+=("-D${sym}=${sym}_64")
+    done
+    export CXXFLAGS="${SYMB_DEFS[@]}"
+fi
+
+if [[ $target == *powerpc* ]]; then
+    FLAGS+=(-DCMAKE_EXE_LINKER_FLAGS="-L/opt/${target}/${target}/lib${arch_ext}/ -lgomp -pthread -ldl")
+fi
+
+if [[ $target == *apple* ]]; then
+    FLAGS+=(-DCMAKE_SYSTEM_NAME="Darwin")
+elif [[ $target == *mingw* ]]; then
+    # We need to remove bigobj support, but we also need to force the linker
+    # to export all symbols because Windows DLLs have the concept of
+    # "exported" and "not exported" functions.
+    sed -i 's/-Wa,-mbig-obj/-Wl,--export-all-symbols/' ../CMakeLists.txt
+
+    # For Windows configuration, we need to manually specify the OpenBLAS
+    # libraries as CMake variables.
+    if [[ "${nbits}" == 64 ]]; then
+        arch_ext="64_";
+    else
+        arch_ext="";
+    fi
+    FLAGS+=(-DLAPACK_LIBRARY=${libdir}/libopenblas${arch_ext}.${dlext})
+    FLAGS+=(-DBLAS_LIBRARY=${libdir}/libopenblas${arch_ext}.${dlext})
+fi
+
+if [[ ${target} != *darwin* ]]; then
+    # Needed to find libgfortran for OpenBLAS.
+    export CXXFLAGS="-Wl,-rpath-link,/opt/${target}/${target}/lib -Wl,-rpath-link,/opt/${target}/${target}/lib64 ${CXXFLAGS}"
+fi
+
+# Reduce number of cores used for builds on FreeBSD or ARM (they run out of
+# memory).
+if [[ $target == *freebsd* ]] || [[ $target == *arm* ]] || [[ $target == *mingw* ]]; then
+  nproc=1;
+fi
+
+cmake .. "${FLAGS[@]}"
+make -j${nproc}
+make install
+
+if [[ ${target} == *mingw* ]]; then
+    cp -v libmlpack_julia*.dll.a "${prefix}/lib"
```

Comment:
Do we really need the static archives in the tarball? :grin:

Reply:
You're right, I think we can remove those... let's see if it still builds and finds all the `LibraryProduct`s right. :)  29b1d9e